src/persist-client/src/internal/state_diff.rs

```diff
@@ -195,6 +195,27 @@ impl<T: Timestamp + Lattice + Codec64> StateDiff<T> {
         }
     }
 
+    pub(crate) fn map_blob_deletes<F: for<'a> FnMut(HollowBlobRef<'a, T>)>(&self, mut f: F) {
+        for spine_diff in self.spine.iter() {
+            match &spine_diff.val {
+                StateFieldValDiff::Insert(()) => {}     // No-op
+                StateFieldValDiff::Update((), ()) => {} // No-op. Should never occur
```

Comment:
Yeah that's a good question, Dan and I debated this one before. It is corruption and means someone wrote invalid data somewhere. My take on Dan's PR back in the day was that crashing the reader here feels a little out of place, since the reader could be far removed from when the bad write occurred, and we should move the validation to the write side instead. But... I'm not 100% attached to that, maybe we should just panic in all these cases after all? (cc @danhhz curious if your opinion has changed at all!)

Reply:
okay yeah I think Dan was right to begin with 😄  I'll turn these into panics here + his original fn